src/ai/backend/manager/scheduler/predicates.py

```diff
@@ -228,19 +236,21 @@ async def _query():
         target_sgroup_names = [preferred_sgroup_name]
     else:
         # Consider all agents in all allowed scaling groups.
+        for sgroup in sgroups:
+            allowed_session_types = sgroup['scheduler_opts']['allowed_session_types']
+            if sess_ctx.session_type.value.lower() in allowed_session_types:
+                break
+        else:
+            return PredicateResult(
+                False,
+                "Not allowed session type in scaling groups.",
+            )
         target_sgroup_names = [sgroup['name'] for sgroup in sgroups]
```

Comment:
This logic should be rewritten to exclude the scaling groups that do not allow the desired session type and report failure *only when there is no scaling group left*.

Reply:
Rewrote and add a test case for the behavior.